public/app/plugins/datasource/loki/responseUtils.ts

```diff
@@ -37,3 +43,25 @@ export function extractHasErrorLabelFromDataFrame(frame: DataFrame): boolean {
   const labels: Array<{ [key: string]: string }> = labelField.values.toArray();
   return labels.some((label) => label['__error__']);
 }
+
+export function extractLevelLikeLabelFromDataFrame(frame: DataFrame): string | undefined {
+  const labelField = frame.fields.find((field) => field.name === 'labels' && field.type === FieldType.other);
+  if (labelField == null) {
+    return undefined;
+  }
+
+  // Depending on number of labels, this can be pretty heavy operation.
+  // Let's just look at first 2 lines If needed, we can introduce more later.
```

Comment:
Because when you get labels, you get all labels. But you want to suggest this only based on labels in your log lines, not all labels.  Moreover, if you use parser, content of your log line becomes label-value pairs and there is currently no API on loki side to get these. So we can only rely on labels in dataframes.

Reply:
Maybe I can explain it better. Imagine that you run a query `{job="grafana"}` that gives you following  log line `foo="bar" baz="foo" lvl="error"`. This log line is in `logfmt` and you can use `logfmt` parser to turn every key-value pair into the label. So if you do `{job="grafana"} | logfmt`, now we also have `lvl` label that could be renamed. But if we wouldn't use parser, this label wouldn't exist.

Reply:
Totally understandable. Thank you for explanation!